### PR TITLE
single global EncodingConfig

### PR DESCRIPTION
**Pull request by jmank88**

### CODE REVIEW

File: pkg/terra/client/client.go [hunk 1]
```diff
@@ -444,3 +442,8 @@ func (c *Client) Balance(addr sdk.AccAddress, denom string) (*sdk.Coin, error) {
 	}
 	return b.Balance, nil
 }
+
+//TODO how much of app.MakeEncodingConfig to reproduce?
```
Comment (jmank88) [inline pkg/terra/client/client.go:446]: `app.MakeEncodingConfig()` is documented for testing only and calls global `Register*` funcs, so we shouldn't use it in production and we can't call it more than once anyways. `params.MakeEncodingConfig` swaps right in, but I don't know if we depend on the global register calls too (and would need to reproduce calling those once from an init).

Comment (jmank88) [reply pkg/terra/client/client.go]: Test failures suggest we need those, so added to a `func init()`.

File: pkg/terra/client/client.go [hunk 2]
```diff
@@ -12,27 +12,42 @@ import (
...
+	encodingConfig := params.MakeEncodingConfig()
+	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
+	std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	app.ModuleBasics.RegisterLegacyAminoCodec(encodingConfig.Amino)
+	app.ModuleBasics.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+
+	// authz module use this codec to get signbytes.
+	// authz MsgExec can execute all message types,
+	// so legacy.Cdc need to register all amino messages to get proper signature
+	app.ModuleBasics.RegisterLegacyAminoCodec(legacy.Cdc)
```
Comment (jmank88) [inline pkg/terra/client/client.go:48]: https://github.com/terra-money/core/blob/main/app/encoding.go#L12-L29

File: pkg/terra/client/client.go [hunk 3]
```diff
@@ -12,27 +12,43 @@ import (
...
 	"github.com/smartcontractkit/terra.go/key"
+	"github.com/smartcontractkit/terra.go/msg"
 	"github.com/smartcontractkit/terra.go/tx"
 )
 
+var encodingConfig = params.MakeEncodingConfig()
+
+func init() {
+	// Based on app.MakeEncodingConfig
+	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
```
Comment (connorwstein) [inline pkg/terra/client/client.go:41]: let's add a comment that these functions can panic when called more than once

**Review by connorwstein — APPROVED** [review]

File: pkg/terra/client/client.go (hunk 3 above)
Comment (jmank88) [reply pkg/terra/client/client.go:41]: :heavy_check_mark:

**Review by connorwstein — APPROVED** [review]

**Pull request merged**